Source/WebCore/dom/NodeRareData.h

```diff
@@ -68,7 +68,8 @@ class NodeListsNodeData {
     void removeChildNodeList(ChildNodeList* list)
     {
         ASSERT(m_childNodeList == list);
-        if (deleteThisAndUpdateNodeRareDataIfAboutToRemoveLastList(list->ownerNode()))
+        Ref ownerNode = list->ownerNode();
+        if (deleteThisAndUpdateNodeRareDataIfAboutToRemoveLastList(ownerNode))
```

Comment:
Why not use `list->protectedOwnerNode()` here?

Reply:
It doesn't exist and we can't define it in ChildNodeList.h because that would require the declaration of Node object, which is only forward declared there.

Reply:
What you did is fine. We could also have:
1. added protectedOwnerNode() and define it out of line if we're not particularly concerned about perf
2. added protectedOwnerNode() and declared it inlined in a new ChildNodeListInlines.h if we're concerned about perf.

Reply:
Yeah, I think we probably want to do (2) since these code paths are relatively hot but I don't think it's worth the cost of adding a new header file for now.